backend/domain/Room.js

```diff
@@ -39,13 +38,29 @@ class Room {
     }
 
     getPlayerByNickname(nickname) {
-        for (player of this.players) {
-            if (player.nickname == nickname) {
-                return player;
+        let returnedPlayer;
+
+        this.players.forEach((player) => {
+            if (player.nickname === nickname) {
+                returnedPlayer = player;
             }
-        }
+        });
+
+        return returnedPlayer;
     }
-    
+
+    getPlayersByRole(role) {
+        const playersOfTheRole = [];
+
+        this.players.forEach((player) => {
+            if (player.role === role) {
+                playersOfTheRole.push(player);
+            }
```

Comment:
need if statement 
of if winning role is civilian, need to include medic and detective
or change the winning role

Reply:
fixed